src/domain/opendatafabric/src/identity/dataset_identity.rs

```diff
@@ -437,6 +476,29 @@ impl DatasetAliasRemote {
     }
 }
 
+impl PartialEq<Self> for DatasetAliasRemote {
+    fn eq(&self, other: &Self) -> bool {
+        (self.repo_name == other.repo_name)
+            && ((self.account_name.is_some()
+                && other.account_name.is_some()
+                && self
+                    .account_name
+                    .as_ref()
+                    .unwrap()
+                    .lowercase_eq(other.account_name.as_ref().unwrap()))
```

Comment:
Try new unstable syntax of `let_chains` Rust feature to implement this. We are already using it in some of our crates:

https://github.com/rust-lang/rfcs/pull/2497

Reply:
Due to lastest changes this block was removed